shared/src/commonMain/kotlin/co/touchlab/kampkit/ktor/DogApiImpl.kt

```diff
@@ -13,7 +13,12 @@ import io.ktor.client.request.HttpRequestBuilder
 import io.ktor.client.request.get
 import io.ktor.http.takeFrom
 
-class DogApiImpl(private val log: Kermit) : KtorApi {
+class DogApiImpl(log: Kermit) : KtorApi {
+
+    // If this is a constructor property, then it gets captured inside HttpClient config and freezes this whole class
+    @Suppress("CanBePrimaryConstructorProperty")
+    private val log = log
```

Comment:
This was the issue I mentioned this morning. Apparently Ktor freezes things inside its configuration blocks starting in 1.4.1. Since we pass the `log` parameter into the `HttpClient` we need to make sure it only captures a constructor parameter rather than a property with a `this` reference. Otherwise, this whole class gets frozen.